app/models/workflow_processor.rb

```diff
@@ -164,14 +157,31 @@ def model_update(step, payload)
     end
   end
 
+  def model_insert(step, payload)
+    resource_type = step["model"]
+    resource = JSON.parse(payload[:resource])
+    post_id = payload["post_id"]
+    
+    attributes = step["attributes"].transform_values do |v|
+      get_value(payload, v).gsub("{{post_id}}", post_id.to_s)
+    end
+  
+    record = resource_type.constantize.create!(attributes)
+    payload["post_id"] = record.id
+    event_data = {
+      insert_type: "base_model",
+      resource_type: resource_type,
+      resource: resource,
+      definition: { attributes: attributes }
+    }
+    Rails.configuration.event_store.publish(
```

Comment:
Same here, only prepare the data and trigger CreateResource event and before that create model_create_handler.rb